Add unlisted to cloud saving options

flxkrmr opened the pull request:
Resolves: #14434

Adds the option to save project as unlisted in the cloud.

I left the "force audio saving" logic untouched, which I think means that it is the same as for private projects. If this needs to be different, let me know

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

Jojo-Schmitz commented:
Not really sure, but I'd rather have "Unlisted" between "Private" and "Public" than after "Public"
(and that merge commit doesn't belong theer, to rebase instead use `git pull --rebase upstream master; git push -f`

shoogle reviewed (commented):
Ideally the options should be specified in this order to match the website: Public, Unlisted, Private.

MueScore.com Score Manager | MuseScore.com Edit Score page
:---:|:---:
![image](https://user-images.githubusercontent.com/11011881/233791096-f96e47df-41d9-42fc-813f-8cf74bf55d0e.png) | ![image](https://user-images.githubusercontent.com/11011881/233791244-43c0a4b2-5416-4120-b9ef-592e72799bc3.png)

You can change this in the `CloudVisibility` enum and in the dialog itself.

The default visibility should be __Private__ for *File > Save to cloud* and __Public__ for *File > Publish to MuseScore.com*.

flxkrmr commented:
The order does make sense, I changed it. I also rebased everything to one commit because I think we don't really need a reorder commit :) However, I just realized in my german gui, unlisted is the only option that is not translated. Should I add a german translation right away and if so, can you point me in a direction how this works?

shoogle commented:
@flxkrmr, you don't need to provide a translation. You just need to mark the source string for translation (which you have done by putting it in the `qsTrc()` function), which will expose it to translators on [Transifex](https://app.transifex.com/musescore/musescore/dashboard/) after this is merged.

However, I have requested the options go in the opposite order to what @Jojo-Schmitz suggested, so please swap them round and double check that the defaults are correct. Sorry for the trouble!

flxkrmr commented:
Thanks for the info.

No problem, I hope the order is correct now :) The default values should also still be correct.

shoogle reviewed (approved):
Thanks for making those changes! I've tested and it's working great!
